File: python/intervals/non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort()
        maximum = intervals[0][1]
        count = 0
        for i in range(1, len(intervals)):
            if intervals[i][0] < maximum:
                count += 1
                maximum = min(maximum, intervals[i][1])
            else:
                maximum = max(maximum, intervals[i][1])

        return count

File: c++/intervals/non-overlapping-intervals.cc
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        std::sort(intervals.begin(), intervals.end());

        int answer = 0;
        int end = std::numeric_limits<int>::min();
        for (std::vector<int> interval : intervals) {
            if (interval[0] < end) {
                answer += 1;
                end = std::min(end, interval[1]);
            } else {
                end = std::max(end, interval[1]);
            }
        }      

        return answer;
    }
};
